### PR TITLE
Consolidate formatting (cleanup + reformat)

### PR DESCRIPTION
**Pull request by clangguth**
PLEASE READ THE FOLLOWING CAREFULLY, AND ONLY MERGE THIS PULL REQUEST WHEN YOU'RE CONFIDENT ABOUT YOUR DECISION

This is an attempt to standardize/normalize/harmonize some aspects which inevitably show up when multiple people with multiple systems work on the same files.

That actually boils down to primarily two things:
 - line endings: CRLF or LF ? 
 - indentation: tabs, spaces, or a mix? (in djigger, it's currently a wild mix between both, sometimes even in a single line :-) )
 
My understanding is that current best practices suggest:
 - using LF as line endings
 - using 4 spaces as indentation for source code
 - using tabs for XML indentation (though I'm not entirely sure about that, it's just what seems to make most sense to me)
 
As these issues are so universal, tools and IDEs also support ways to help with configuring these settings.
Namely:
 - git supports a .gitattributes file for configuring line break behavior: https://help.github.com/articles/dealing-with-line-endings/
 - many IDEs support EditorConfig specifications: http://editorconfig.org/
 
This pull request adds two configuration files (.gitattributes and .editorconfig), with what I think are sensible settings according to best practices, and all source files have been reformatted accordingly.

It was created by running IntelliJ IDEA "Reformat code" on the entire project, and should only contain purely cosmetical changes. In other words:
- Java files have been reformatted with normalized line endings, normalized spaces, and optimized imports. Other minor changes may also have been introduced by the formatting rules (like "a==b" becoming "a == b")
- XML files have been reformatted with normalized line endings and normalized spaces

Again, while the changes affect almost the entire project (because most lines are touched), they should not have any effect whatsoever in terms of functionality.
They were only made to ensure that future collaborative editing gets easier, because most of the formatting issues should then be out of the way.

I have tested this as far as I could, and I'm confident that everything is correct (and of course, I'd like to see this merged :-) ).

However, please actually check if everything is still working for you too, and if you're happy with the results.

In a nutshell, you could test it like this:

```
git remote add clangguth https://github.com/clangguth/djigger.git
git fetch clangguth
git checkout --track clangguth/cleanup_reformat
```

If everything works for you too, and if you're OK with the suggested changes, then feel free to merge this PR. Otherwise, I'm open to further comments and suggestions, and looking forward to your feedback.

Thanks & cheers :-)
Chris

### CODE REVIEW

**Comment by clangguth** [comment]
Hi folks,

any feedback on this? I'd like to go ahead with #98, but IMO it makes sense to merge this PR before I do so.

Thanks!
Chris

**Comment by dcransac** [comment]
We're checking this. I'll give you an update shortly.

**Comment by jeromecomte** [comment]
Hi Chris,

Thanks a lot. We've never noticed this before! Maybe because we're all using the default eclipse formatter which displays tabs with the size of 4 spaces (and apparently different indentation policies).

Generally we try to avoid this kind of formatting changes to ease the diffs between commits/branches/versions.

As we're currently in a quite stable phase it might however be the right time to undertake such a change.

I'll therefore trust your IDE... and make sure we all install the EditorConfig Plugin for eclipse to avoid these problems in the future.

Again many thanks and sorry for the delay,
Jérôme

**Comment by clangguth** [comment]
Hi Jérôme, Dorian,

awesome!

Thanks for merging, and btw thanks a lot for the great software :-)
